File: android/InverterMonitor/app/src/main/java/com/bilalahmad/invertermonitor/ui/components/StatusPill.kt
package com.bilalahmad.invertermonitor.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.bilalahmad.invertermonitor.ui.theme.Palette

@Composable
fun StatusPill(
    label: String,
    icon: ImageVector? = null,
    tint: Color = Color.White,
    backgroundTint: Color = Palette.CardSurface,
    dotColor: Color? = null,
    dashed: Boolean = false,
    modifier: Modifier = Modifier,
) {
    val shape = RoundedCornerShape(999.dp)
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        modifier = modifier
            .clip(shape)
            .background(backgroundTint)
            .border(
                BorderStroke(1.dp, Palette.CardBorder.copy(alpha = if (dashed) 0.6f else 1f)),
                shape
            )
            .padding(horizontal = 10.dp, vertical = 5.dp)
    ) {
        if (dotColor != null) {
            Box(
                modifier = Modifier
                    .size(8.dp)
                    .clip(CircleShape)
                    .background(dotColor)
            )
        }
        if (icon != null) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = tint,
                modifier = Modifier.size(14.dp),
            )
        }
        Text(
            text = label,
            fontSize = 12.sp,
            fontWeight = FontWeight.SemiBold,
            color = tint,
        )
    }
}

File: ios/InverterMonitor/InverterMonitor/Views/Shared/StatusPill.swift
import SwiftUI

struct StatusPill: View {
    let label: String
    var systemImage: String? = nil
    var tint: Color = .white
    var backgroundTint: Color = Palette.cardSurface
    var dotColor: Color? = nil
    var dashed: Bool = false

    var body: some View {
        HStack(spacing: 6) {
            if let dotColor {
                Circle()
                    .fill(dotColor)
                    .frame(width: 8, height: 8)
            }
            if let systemImage {
                Image(systemName: systemImage)
                    .imageScale(.small)
                    .foregroundStyle(tint)
            }
            Text(label)
                .font(.caption.weight(.semibold))
                .foregroundStyle(tint)
        }
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(
            Capsule(style: .continuous)
                .fill(backgroundTint)
        )
        .overlay(
            Capsule(style: .continuous)
                .strokeBorder(
                    Palette.cardBorder,
                    style: StrokeStyle(lineWidth: 1, dash: dashed ? [3, 3] : [])
                )
        )
    }
}
